### ISSUE TITLE
How performant is Dripsy?

### ISSUE THREAD

**Issue by PrimulaX** [nandorojo/dripsy#308]
I just found some ui libs benchmarks and it looks like dripsy has the worst performance: https://github.com/efstathiosntonas/react-native-style-libraries-benchmark

Is dripsy that slow or just some other things going on there?

**Comment by nandorojo** [comment]
weird, we haven’t gotten a real-life performance issue before. if it gets isolated down to some part of the code i’m happy to merge a fix

**Comment by PrimulaX** [comment]
@nandorojo That's the code they are using, I tried to check by myself in Profiler and it is quite true, this component here takes a lot of time to render
![image](https://github.com/nandorojo/dripsy/assets/16132259/e5f76bdf-fec7-435c-8303-2893b98f6198)

**Comment by anatooly** [comment]
> That's the code they are using, I tried to check by myself in Profiler and it is quite true, this component here takes a lot of time to render !

First place of problem is value superStyle (src/core/css/map-props.ts)
```
  // const superStyle = css(sx, breakpoint)({ theme, fontFamily, themeKey })
  const superStyle = {}
 ```

LOG  baseStyle undefined 0 {} undefined undefined
LOG  superStyle {"borderColor": "red", "borderWidth": 2, "padding": 5} 0 {} undefined undefined
 
 if not calculate it perfomance change from ~1300 to ~300 like all other framework on test.
 
/src/core/css/index.tsx main prolem on cycle "for" function export const css()
  `for (const key in styles) {`
  
on testcase create View elements, nothing code into `for` cycle not needed but it execute, time 1400 -> 450 if disable all inside check.
  
  input: {"borderColor": "red", "borderWidth": 2, "padding": 5}
  output: {"borderColor": "red", "borderWidth": 2, "padding": 5}
  
  I think problem is merge two object on this place:
  ```
  const themedAliases = {
      ...aliases,
     ...theme.aliases,
     }
 ``` 
 or refactor code inside `for (const key in styles) {`

I add some pull request https://github.com/nandorojo/dripsy/pull/310

**Comment by nandorojo** [comment]
@anatooly Hey, sorry I missed this. Just added an additional perf fix to that PR and merged it

I'm sure there are tons of perf wins from small changes of this kind, namely for creating fewer objects and spreads where it's not needed. I think this may have become a bigger issue in Hermes over JSC

**Comment by nandorojo** [comment]
Published in 4.3.6. 

I don't have a _ton_ of time right now but if there are more perf wins that improve the benchmarks for this, I'm happy to merge them. Feel free to tag me if you find any. I'd prefer small changes like this over big refactors if possible.

**Comment by nandorojo** [comment]
Added some more instances of `Object.assign` over spreads in `4.3.7`:

https://github.com/nandorojo/dripsy/commit/8e89b4b8c2bcf66be731c25dc67f32eaac0d0e58

**Comment by PrimulaX** [comment]
Hey @nandorojo, it looks like you haven’t released it yet

**Comment by nandorojo** [comment]
it's published on NPM, see 4.3.8 too

**Comment by PrimulaX** [comment]
So these changes made performance improvement, it was around 650ms <img width="402" alt="image" src="https://github.com/user-attachments/assets/c8ed37e7-3f34-4bab-94eb-afc3165e423b">
